File: frontend/src/services/api/resume.ts
import { apiClient } from './client';

export interface ResumeSuggestion {
  id: string;
  resume_id: string;
  section: string;
  suggestion_type: string;
  current_text: string;
  recommended_text: string;
  impact_reason?: string;
  is_applied: boolean;
  created_at: string;
}

export interface Resume {
  id: string;
  user_id: string;
  title: string;
  target_role: string;
  raw_text?: string;
  parsed_sections: {
    summary?: string;
    experience?: string[];
    skills?: string[];
  };
  ats_score: number;
  created_at: string;
  updated_at: string;
  suggestions: ResumeSuggestion[];
}

export const resumeApi = {
  list: async (): Promise<Resume[]> => {
    return apiClient.get('/resume');
  },
  create: async (data: { title: string; target_role: string; raw_text?: string }): Promise<Resume> => {
    return apiClient.post('/resume', data);
  },
  getById: async (id: string): Promise<Resume> => {
    return apiClient.get(`/resume/${id}`);
  },
  applySuggestion: async (resumeId: string, suggestionId: string): Promise<Resume> => {
    return apiClient.post(`/resume/${resumeId}/suggestions/${suggestionId}/apply`, { apply: true });
  },
};

File: backend/app/models/resume.py
from sqlalchemy import Column, String, Text, DateTime, ForeignKey, Integer, Float, JSON, Boolean, Uuid
from sqlalchemy.orm import relationship
from datetime import datetime, timezone
import uuid
from app.core.database import Base

class Resume(Base):
    __tablename__ = "resumes"

    id = Column(Uuid, primary_key=True, default=uuid.uuid4)
    user_id = Column(Uuid, ForeignKey("users.id", ondelete="CASCADE"), nullable=False, index=True)
    title = Column(String(255), nullable=False)
    target_role = Column(String(255), nullable=False)
    raw_text = Column(Text, nullable=True)
    parsed_sections = Column(JSON, default=dict)  # {experience: [], skills: [], education: []}
    ats_score = Column(Float, default=70.0)
    created_at = Column(DateTime, default=lambda: datetime.now(timezone.utc))
    updated_at = Column(DateTime, default=lambda: datetime.now(timezone.utc), onupdate=lambda: datetime.now(timezone.utc))

    suggestions = relationship("ResumeSuggestion", back_populates="resume", cascade="all, delete-orphan", lazy="selectin")

class ResumeSuggestion(Base):
    __tablename__ = "resume_suggestions"

    id = Column(Uuid, primary_key=True, default=uuid.uuid4)
    resume_id = Column(Uuid, ForeignKey("resumes.id", ondelete="CASCADE"), nullable=False, index=True)
    section = Column(String(100), nullable=False)  # experience, summary, skills
    suggestion_type = Column(String(50), nullable=False)  # IMPACT_BULLET, MISSING_KEYWORD, QUANTIFIABLE_METRIC
    current_text = Column(Text, nullable=False)
    recommended_text = Column(Text, nullable=False)
    impact_reason = Column(Text, nullable=True)
    is_applied = Column(Boolean, default=False)
    created_at = Column(DateTime, default=lambda: datetime.now(timezone.utc))

    resume = relationship("Resume", back_populates="suggestions")
